microsoft/testsuites/performance/common.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from typing import Dict, List, Optional
+
+from lisa import Node, notifier
+from lisa.environment import Environment
+from lisa.notifier import DiskPerformanceMessage, DiskSetupType, DiskType
+from lisa.tools import FIOMODES, Fio, FIOResult
+from lisa.util import dict_to_fields
+
+
+def run_perf_test(
+    node: Node,
+    start_qdepth: int,
+    max_qdepth: int,
+    filename: str,
+    num_jobs: Optional[List[int]] = None,
+    block_size: int = 4,
+    size_gb: int = 0,
+    numjob: int = 0,
+    overwrite: bool = False,
+) -> List[DiskPerformanceMessage]:
+    fio_result_list: List[FIOResult] = []
+    fio = node.tools[Fio]
+    numjobiterator = 0
+    for mode in FIOMODES:
+        qdepth = start_qdepth
+        while qdepth <= max_qdepth:
```

Comment:
The loop should be changed to check the iodepth. The qdepth will be a value only, not use it in any calculation.

Reply:
then in future we can abandon qdepth